pandas/core/arrays/period.py

```diff
@@ -360,7 +354,12 @@ def _add_offset(self, other):
         if base != self.freq.rule_code:
             msg = DIFFERENT_FREQ_INDEX.format(self.freqstr, other.freqstr)
             raise IncompatibleFrequency(msg)
-        return self._time_shift(other.n)
```

Comment:
Was this doing the wrong thing? (to understand the reason of the change)

Reply:
```
pi = pd.PeriodIndex(['2016Q1', '2016Q2'], freq='2Q')
pi + pi.freq
PeriodIndex(['2017Q1', '2017Q2'], dtype='period[2Q-DEC]', freq='2Q-DEC')
```

i.e. it is adding 4Q instead of 2Q.

Reply:
Shouldn't we then fix `_time_shift`? Or has it other use cases for which it is working correctly?

Reply:
_time_shift itself is correct.  The issue is add_offset passing the wrong argument to time_shift

Reply:
You are being a bit cryptic here (why then not passing the correct argument?)

Reply:
```
pi = pd.PeriodIndex(['2016Q1', '2016Q2'], freq=str(n) + 'Q'))
pi + k * pi.freq
```

In master this passes `k * n` to time_shift, where the correct argument is just `k`